ignite/metrics/multilabel_confusion_matrix.py

```diff
@@ -0,0 +1,132 @@
+        M[i, 0, 1] corresponds to count/rate of false positives of class i,
+        M[i, 1, 0] corresponds to count/rate of false negatives of class i,
+        M[i, 1, 1] corresponds to count/rate of true positives of class i.
+    - The classes present in M are indexed as 0, ..., num_classes-1 as can be inferred from above.
+
+    Args:
+        num_classes (int): Number of classes, should be > 1. See notes for more details.
+        output_transform (callable, optional): a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        device (str or torch.device): specifies which device updates are accumulated on. Setting the metric's
+            device to be the same as your ``update`` arguments ensures the ``update`` method is non-blocking. By
+            default, CPU.
+
+    """
+
+    def __init__(
+        self,
+        num_classes: int,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+        normalized: bool = False,
+    ):
+        if num_classes <= 1:
+            raise ValueError("Argument num_classes needs to be > 1")
+
+        self.num_classes = num_classes
+        self._num_examples = 0
+        self.normalized = normalized
+        super(MultiLabelConfusionMatrix, self).__init__(output_transform=output_transform, device=device)
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self.confusion_matrix = torch.zeros(self.num_classes, 2, 2, dtype=torch.int64, device=self._device)
+        self._num_examples = 0
+
+    @reinit__is_reduced
+    def update(self, output: Sequence[torch.Tensor]) -> None:
+        self._check_input(output)
+        y_pred, y = output[0].detach(), output[1].detach()
+
+        self._num_examples += y.shape[0]
+        y_reshaped = y.transpose(0, 1).reshape(self.num_classes, -1)
+        y_pred_reshaped = y_pred.transpose(0, 1).reshape(self.num_classes, -1)
+
+        y_total = y_reshaped.sum(dim=1)
+        y_pred_total = y_pred_reshaped.sum(dim=1)
+
+        tp = (y_reshaped * y_pred_reshaped).sum(dim=1)
+        fp = y_pred_total - tp
+        fn = y_total - tp
+        tn = y_reshaped.shape[1] - tp - fp - fn
+
+        self.confusion_matrix += torch.stack([tn, fp, fn, tp], dim=1).reshape(-1, 2, 2).to(self._device)
+
+    @sync_all_reduce("confusion_matrix", "_num_examples")
+    def compute(self) -> torch.Tensor:
+        if self._num_examples == 0:
+            raise NotComputableError("Confusion matrix must have at least one example before it can be computed.")
+
+        if self.normalized:
+            conf = self.confusion_matrix.to(dtype=torch.float64)
+            sums = conf.sum(dim=(1, 2))
+            return conf / sums[:, None, None]
```

Comment:
Seems like there is no test for normalized CM ?

Reply:
I can add that in the next commit along with the new distributed settings test. Since the normalization feature is independent from the shapes of y and y_pred, I think it suffices to add similar tests as in ```test_simple_2D_input``` using ```normalized=True```

Reply:
>  I think it suffices to add similar tests as in test_simple_2D_input using normalized=True

Sounds good ! We can even refactor existing test_simple_2D_input as 
```python

def test_simple_2D_input():

    def _test(normalized):
         # all previous code with adaptation for normalized

    _test(True)
    _test(False)

```